db_stress_tool/db_stress_driver.cc

```diff
@@ -81,31 +83,6 @@ bool RunStressTestImpl(SharedState* shared) {
   stress->InitDb(shared);
   stress->FinishInitDb(shared);
 
-  if (FLAGS_write_fault_one_in) {
-    if (!FLAGS_sync_fault_injection) {
-      // unsynced WAL loss is not supported without sync_fault_injection
-      fault_fs_guard->SetDirectWritableTypes({kWalFile});
```

Comment:
I found out we don't need to do this for below reasoning.

(1) `FLAGS_write_fault_one_in` and `FLAGS_sync_fault_injection` both rely  on `TestFSWritableFile`. Loss of un-synced data and write error injection are all implemented in `TestFSWritableFile` and once `TestFSWritableFile`is used, you can't choose to have write error injection but not loss of un-synced data. 

(2) So when `FLAGS_write_fault_one_in` and since now we include WAL file, there can be loss of unsynced WAL data. 

(3) Because of (2), I instructed the crash recovery to use prefix recovery below (see function `MightHaveDataLoss()`) once `FLAGS_write_fault_one_in` is used. So even when `FLAGS_sync_fault_injection` is false, we support un-synced WAL loss.